services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/oss/OssUtil.java

```diff
@@ -0,0 +1,253 @@
+        List<SearchFacetOption> searchFacetOptions = searchQuery.getFacetOptions();
+        Set<Long> userGroups = getAuthorizedUserGroups(userInfo);
+        Map<String, Aggregation> aggregations = new HashMap<>();
+
+        BoolQuery.Builder boolBuilder = new BoolQuery.Builder();
+        SearchRequest.Builder searchBuilder = new SearchRequest.Builder()
+                .index(SearchConstants.OPEN_SEARCH_INDEX_NAME);
+
+        // clean up empty q
+        if (terms != null && terms.size() == 1 && "".equals(terms.get(0))) {
+            terms = null;
+        }
+
+        // check for minimum search requirements
+        if (CollectionUtils.isEmpty(terms) && CollectionUtils.isEmpty(booleanQueries)) {
+            throw new IllegalArgumentException(
+                    "Either one queryTerm or one booleanQuery must be defined.");
+        }
+
+        if (!CollectionUtils.isEmpty(terms)) {
+            String queryTerms = String.join(" ", terms);
+            boolBuilder.must(
+                    MultiMatchQuery.of(m -> m
+                            .query(queryTerms)
+                    )._toQuery()
+                    //You can specify the behavior for an empty query in the zero_terms_query parameter.
+                    // Setting zero_terms_query to all returns all documents in the index and setting it to none returns no documents:
+            );
+        } else {
+            // If no terms, use match_all
+            boolBuilder.must(
+                    MatchAllQuery.of(m -> m)._toQuery()
+            );
+        }
+
+        if (!CollectionUtils.isEmpty(booleanQueries)) {
+            booleanQueries.forEach(query -> {
+                Query termQuery = TermQuery.of(t -> t
+                        .field(query.getKey())
+                        .value(FieldValue.of(query.getValue()))
+                )._toQuery();
+                if (query.getNot() != null && query.getNot()) {
+                    // Add to must_not if this is a NOT condition
+                    boolBuilder.mustNot(termQuery);
+                } else {
+                    boolBuilder.must(termQuery);
+                }
+            });
+        }
+
+        if (!CollectionUtils.isEmpty(rangeQueries)) {
+
+            rangeQueries.forEach(query -> {
+                ValidateArgument.required(query.getKey(), "keyRange key");
+                if (query.getMax() == null && query.getMin() == null) {
+                    throw new IllegalArgumentException("At least one of min or max for key=" + query.getKey() + " must be not null.");
+                }
+
+                RangeQuery.Builder builder = new RangeQuery.Builder().field(query.getKey());
+                if (query.getMin() != null){
+                    builder.gte(JsonData.of(query.getMin()));
+                }
+                if (query.getMax() != null) {
+                    builder.lte(JsonData.of(query.getMax()));
+                }
```

Comment:
is the old query inclusive or exclusive?  This will answer if it should be gte or gt.